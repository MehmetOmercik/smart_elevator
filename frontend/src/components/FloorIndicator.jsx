import React, { useEffect, useState } from "react";
import { SlArrowUpCircle, SlArrowDownCircle } from "react-icons/sl";
import { useSelector } from "react-redux";

export default function FloorIndicator(props) {
	const floorRequest = useSelector((state) => state.button.floor);
	const floorDifference = useSelector(
		(state) => state.button.floorDifference
	);
	const currentFloor = useSelector((state) => state.button.currentFloor);
	const oldFloor = floorRequest - floorDifference;
	const elevator = useSelector((state) => state.button.elevator);
	let colorUp;
	let colorDown;

	if (floorDifference > 0 && currentFloor !== floorRequest) {
		colorUp = "green";
	} else if (floorDifference < 0 && currentFloor !== floorRequest) {
		colorDown = "red";
	} else {
		colorUp = "";
	}

	return (
		<div className='floorindicator-container'>
			<p className='floorindicator-container__name'>
				Elevator NO: {props.elevatorid}
			</p>
			<span>{elevator === props.elevatorid ? currentFloor : 0}</span>
			<SlArrowUpCircle
				color={elevator === props.elevatorid ? colorUp : ""}
				fontSize='3rem'
			/>
			<SlArrowDownCircle
				color={elevator === props.elevatorid ? colorDown : ""}
				fontSize='3rem'
			/>
		</div>
	);
}
