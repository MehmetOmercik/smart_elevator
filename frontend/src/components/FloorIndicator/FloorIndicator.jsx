import React, { useEffect, useState } from "react";
import { SlArrowUpCircle, SlArrowDownCircle } from "react-icons/sl";
import { useSelector } from "react-redux";

export default function FloorIndicator(props) {
  const floorRequest = useSelector((state) => state.button.floor);
  const floorDifference = useSelector((state) => state.button.floorDifference);
  const currentFloor = useSelector((state) => state.button.currentFloor);
  const oldFloor = floorRequest - floorDifference;
  const elevatorID = useSelector((state) => state.button.elevatorID);
  let colorUp;
  let colorDown;

  if (floorDifference > 0 && currentFloor !== floorRequest) {
    colorUp = "green";
  } else if (floorDifference < 0 && currentFloor !== floorRequest) {
    colorDown = "red";
  } else {
    colorUp = "";
  }

  return (
    <div className="floorindicator-container">
      <p className="floorindicator-container__name">
        Elevator NO: {props.elevatorid}
      </p>
      <span data-testid="floor-number">
        {elevatorID === props.elevatorid ? currentFloor : 0}
      </span>
      <SlArrowUpCircle
        data-testid="up-arrow"
        color={elevatorID === props.elevatorid ? colorUp : ""}
        fontSize="3rem"
      />
      <SlArrowDownCircle
        data-testid="down-arrow"
        color={elevatorID === props.elevatorid ? colorDown : ""}
        fontSize="3rem"
      />
    </div>
  );
}
