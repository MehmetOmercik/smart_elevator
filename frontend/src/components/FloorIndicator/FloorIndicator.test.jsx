import { render, screen } from "@testing-library/react";
import FloorIndicator from "./FloorIndicator";
import { renderWithProviders } from "../../test/test-utils";

describe("Floor Indicator tests", () => {
  test("snapshot", () => {
    const floorIndicator = renderWithProviders(<FloorIndicator />);
    expect(floorIndicator).toMatchSnapshot();
  });

  test("elevator name rendering", () => {
    renderWithProviders(<FloorIndicator />);
    const elevatorId = screen.getByText(/Elevator NO:/i);
    expect(elevatorId).toBeInTheDocument();
  });

  test("up arrow rendering", () => {
    const { getByTestId } = renderWithProviders(<FloorIndicator />);
    const upArrow = getByTestId("up-arrow");
    expect(upArrow).toBeVisible();
  });

  test("down arrow rendering", () => {
    const { getByTestId } = renderWithProviders(<FloorIndicator />);
    const downArrow = getByTestId("down-arrow");
    expect(downArrow).toBeInTheDocument();
  });

  test("floor number rendering", () => {
    const { getByTestId } = renderWithProviders(<FloorIndicator />);
    const floorNumber = getByTestId("floor-number");
    expect(floorNumber).toBeInTheDocument();
  });
});
