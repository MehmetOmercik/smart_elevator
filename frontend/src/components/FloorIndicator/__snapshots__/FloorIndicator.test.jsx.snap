// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Floor Indicator tests > snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="floorindicator-container"
      >
        <p
          class="floorindicator-container__name"
        >
          Elevator NO: 
        </p>
        <span
          data-testid="floor-number"
        />
        <svg
          color=""
          data-testid="up-arrow"
          fill="currentColor"
          font-size="3rem"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1024 512C1024 229.216 794.768 0 512 0 229.216 0 0 229.216 0 512c0 282.768 229.216 512 512 512 282.768 0 512-229.232 512-512zm-960.992 0C63.008 264.976 264.976 64 512 64c247.024 0 448 200.976 448 448S759.024 960 512 960 63.008 759.024 63.008 512zm481.328 224.32V375.856l115.76 115.76c12.496 12.496 32.752 12.496 45.248 0s12.496-32.752 0-45.248l-194-189.008-194 189.008c-6.256 6.256-9.376 14.432-9.376 22.624s3.12 16.368 9.376 22.624c12.496 12.496 32.752 12.496 45.248 0l117.744-117.76V736.32c0 17.68 14.336 32 32 32s32-14.32 32-32z"
          />
        </svg>
        <svg
          data-testid="down-arrow"
          fill="currentColor"
          font-size="3rem"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 512c0 282.784 229.232 512 512 512 282.784 0 512-229.216 512-512C1024 229.232 794.784 0 512 0 229.232 0 0 229.232 0 512zm961.008 0c0 247.024-201.969 448-449.009 448s-448-200.976-448-448 200.976-448 448-448 449.009 200.976 449.009 448zM479.663 287.68v360.448l-115.76-115.76c-12.496-12.496-32.752-12.496-45.248 0s-12.496 32.752 0 45.248l194.016 189.008 194-189.008c6.256-6.256 9.376-14.432 9.376-22.624s-3.12-16.368-9.376-22.624c-12.496-12.496-32.752-12.496-45.248 0l-117.744 117.76V287.68c0-17.68-14.336-32-32-32s-32.016 14.32-32.016 32z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="floorindicator-container"
    >
      <p
        class="floorindicator-container__name"
      >
        Elevator NO: 
      </p>
      <span
        data-testid="floor-number"
      />
      <svg
        color=""
        data-testid="up-arrow"
        fill="currentColor"
        font-size="3rem"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1024 512C1024 229.216 794.768 0 512 0 229.216 0 0 229.216 0 512c0 282.768 229.216 512 512 512 282.768 0 512-229.232 512-512zm-960.992 0C63.008 264.976 264.976 64 512 64c247.024 0 448 200.976 448 448S759.024 960 512 960 63.008 759.024 63.008 512zm481.328 224.32V375.856l115.76 115.76c12.496 12.496 32.752 12.496 45.248 0s12.496-32.752 0-45.248l-194-189.008-194 189.008c-6.256 6.256-9.376 14.432-9.376 22.624s3.12 16.368 9.376 22.624c12.496 12.496 32.752 12.496 45.248 0l117.744-117.76V736.32c0 17.68 14.336 32 32 32s32-14.32 32-32z"
        />
      </svg>
      <svg
        data-testid="down-arrow"
        fill="currentColor"
        font-size="3rem"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 512c0 282.784 229.232 512 512 512 282.784 0 512-229.216 512-512C1024 229.232 794.784 0 512 0 229.232 0 0 229.232 0 512zm961.008 0c0 247.024-201.969 448-449.009 448s-448-200.976-448-448 200.976-448 448-448 449.009 200.976 449.009 448zM479.663 287.68v360.448l-115.76-115.76c-12.496-12.496-32.752-12.496-45.248 0s-12.496 32.752 0 45.248l194.016 189.008 194-189.008c6.256-6.256 9.376-14.432 9.376-22.624s-3.12-16.368-9.376-22.624c-12.496-12.496-32.752-12.496-45.248 0l-117.744 117.76V287.68c0-17.68-14.336-32-32-32s-32.016 14.32-32.016 32z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
